File: tests/selenium/pages/LoginPage.js
// Login Page Object Model
const { By } = require('selenium-webdriver');
const config = require('../config');

class LoginPage {
  constructor(driver) {
    this.driver = driver;
    
    // Locators
    this.emailInput = By.css('input[type="email"], input[name="email"]');
    this.passwordInput = By.css('input[type="password"], input[name="password"]');
    this.loginButton = By.css('button[type="submit"]');
    this.errorMessage = By.css('.error, [class*="error"], .alert-danger');
    this.successMessage = By.css('.success, [class*="success"], .alert-success');
  }

  async navigate() {
    await this.driver.get(config.BASE_URL);
  }

  async login(email, password) {
    await this.driver.findElement(this.emailInput).sendKeys(email);
    await this.driver.findElement(this.passwordInput).sendKeys(password);
    await this.driver.findElement(this.loginButton).click();
  }

  async isErrorDisplayed() {
    try {
      const element = await this.driver.findElement(this.errorMessage);
      return await element.isDisplayed();
    } catch (error) {
      return false;
    }
  }

  async getErrorMessage() {
    try {
      const element = await this.driver.findElement(this.errorMessage);
      return await element.getText();
    } catch (error) {
      return '';
    }
  }

  async isLoginSuccessful() {
    await this.driver.sleep(2000);
    const currentUrl = await this.driver.getCurrentUrl();
    return !currentUrl.includes('login');
  }
}

module.exports = LoginPage;

File: selenium-tests/src/main/java/com/yatrik/erp/pages/LoginPage.java
package com.yatrik.erp.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

/**
 * Page Object Model for Login Page
 */
public class LoginPage extends BasePage {
    
    // Locators - try multiple selectors
    private By emailInput = By.id("email");
    private By passwordInput = By.id("password");
    private By loginButton = By.cssSelector("button[type='submit']");
    private By errorMessage = By.cssSelector(".error-message, .alert-danger");
    private By registerLink = By.linkText("Register");
    private By forgotPasswordLink = By.linkText("Forgot Password");

    public LoginPage(WebDriver driver) {
        super(driver);
        // Wait longer for React to fully render the login form
        System.out.println("⏳ Waiting for login page to load...");
        try {
            Thread.sleep(3000); // Give React more time
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }

    public void enterEmail(String email) {
        System.out.println("📧 Entering email...");
        type(emailInput, email);
    }

    public void enterPassword(String password) {
        System.out.println("🔒 Entering password...");
        type(passwordInput, password);
    }

    public void clickLoginButton() {
        System.out.println("🖱️  Clicking login button...");
        click(loginButton);
    }

    public void login(String email, String password) {
        enterEmail(email);
        enterPassword(password);
        clickLoginButton();
    }

    public boolean isErrorMessageDisplayed() {
        return isDisplayed(errorMessage);
    }

    public String getErrorMessage() {
        return getText(errorMessage);
    }

    public boolean isLoginPageDisplayed() {
        return isDisplayed(emailInput) && isDisplayed(passwordInput);
    }
}
